openpower/package/hostboot/0001-loadPnorLayout-accepts-optional-parameter-to-place.patch

```diff
@@ -0,0 +1,115 @@
+From 12063f8d5e0bac864d35c2595a0ae0942f151bc0 Mon Sep 17 00:00:00 2001
+From: Luis Fernandez <Luis.Fernandez@ibm.com>
+Date: Fri, 18 Oct 2019 15:53:23 -0500
+Subject: [PATCH] loadPnorLayout accepts optional parameter to place edited xml
+ layout file.
+
+Change-Id: I7b9ee0f7ad33ad88be58598ddf594765eabe29ea
+---
+ src/build/buildpnor/PnorUtils.pm     | 17 ++++++++++++++---
+ src/build/buildpnor/buildpnor.pl     |  8 +++++++-
+ src/build/buildpnor/genPnorImages.pl |  5 ++++-
+ 3 files changed, 25 insertions(+), 5 deletions(-)
+
+diff --git a/src/build/buildpnor/PnorUtils.pm b/src/build/buildpnor/PnorUtils.pm
+index 7011910..298fe06 100644
+--- a/src/build/buildpnor/PnorUtils.pm
++++ b/src/build/buildpnor/PnorUtils.pm
+@@ -54,7 +54,8 @@ use constant PAGE_SIZE => 4096;
+ ################################################################################
+ sub loadPnorLayout
+ {
+-    my ($i_pnorFile, $i_pnorLayoutRef, $i_physicalOffsets, $i_testRun) = @_;
++    my ($i_pnorFile, $i_pnorLayoutRef, $i_physicalOffsets, $i_testRun,
++        $i_outputLayoutLocation) = @_;
+     my $this_func = (caller(0))[3];
+ 
+     unless(-e $i_pnorFile)
+@@ -255,9 +256,19 @@ sub loadPnorLayout
+         checkForOverlap($i_pnorLayoutRef);
+     }
+ 
+-    # write xml with offsets to new file
++    # Write xml with offsets to new file
++
+     my $filename = basename($i_pnorFile, ".xml");
+-    $filename = "${filename}WithOffsets.xml";
++
++    if (defined $i_outputLayoutLocation)
++    {
++        $filename = "${i_outputLayoutLocation}/${filename}WithOffsets.xml";
++    }
++    else {
++        # If argument $i_outputLayoutLocation is not supplied, the new layout
```

Comment:
I think it is better to do not create the file at all, if output location is not specified.

Reply:
Agree, made changes to reflect that.